pkg/syncer/spec/spec_process.go

```diff
@@ -174,11 +174,23 @@ func (c *Controller) process(ctx context.Context, gvr schema.GroupVersionResourc
 	}
 
 	// upsert downstream
-	u, ok := obj.(*unstructured.Unstructured)
+	upstreamObj, ok := obj.(*unstructured.Unstructured)
 	if !ok {
 		return fmt.Errorf("object to synchronize is expected to be Unstructured, but is %T", obj)
 	}
-	return c.applyToDownstream(ctx, gvr, downstreamNamespace, u)
+
+	if err := c.ensureDownstreamNamespaceExists(ctx, downstreamNamespace, upstreamObj); err != nil {
+		return err
+	}
+
+	if addedFinalizerInUpstream, err := c.ensureSyncerFinalizer(ctx, gvr, upstreamObj); addedFinalizerInUpstream {
```

Comment:
nit: `added` is enough. There is enough context to understand.

Reply:
Fixed